staging/src/k8s.io/legacy-cloud-providers/azure/azure.go

```diff
@@ -271,6 +275,19 @@ func NewCloud(configReader io.Reader) (cloudprovider.Interface, error) {
 		unmanagedNodes:     sets.NewString(),
 		routeCIDRs:         map[string]string{},
 	}
+	func() {
+		// this allows the code ot launch without featuregates defined.  It is currently required for unit tests where the
+		// featuregates are not registered.  This is effectively coding by side effect and an explicit register should
+		// be eventually created instead.
+		defer func() {
```

Comment:
If you split the NewCloud function so that test code doesn't have to execute this, do you still need to catch the panic?

Reply:
> If you split the NewCloud function so that test code doesn't have to execute this, do you still need to catch the panic?

And let it straight die instead?  That's not consistent with existing behavior. I'm honestly unsure if this ever worked properly when it really ran.  Are you sure you want to change behavior?

Reply:
> And let it straight die instead?

Yup, that's the goal of the change, isn't it? :)